sympy/physics/mechanics/kane.py

```diff
@@ -697,20 +697,32 @@ def kanes_equations(self, FL, BL):
         Parameters
         ==========
 
-        FL : list
-            Takes in a list of (Point, Vector) or (ReferenceFrame, Vector)
+        bodies : iterable
+            An iterable of all RigidBody's and Particle's in the system.
+            A system must have at least one body.
+        loads : iterable
+            Takes in an iterable of (Particle, Vector) or (ReferenceFrame, Vector)
             tuples which represent the force at a point or torque on a frame.
-        BL : list
-            A list of all RigidBody's and Particle's in the system.
-
+            Must be either a non-empty iterable of tuples or None which corresponds
+            to a system with no constraints.
         """
+        if (bodies is None and loads != None) or isinstance(bodies[0], tuple):
+            # This switches the order if they use the old way.
+            bodies, loads = loads, bodies
+            SymPyDeprecationWarning(value='The API for kanes_equations() has changed such '
+                    'that the loads (forces and torques) are now the second argument '
+                    'and is optional with None being the default.',
+                    feature='The kanes_equation() argument order',
+                    last_supported_version="1.0",
+                    useinstead='switched argument order to update your code, For example: '
+                    'kanes_equations(loads, bodies) > kanes_equations(bodies, loads).',
+                    issue=10945, deprecated_since_version="1.0").warn()
```

Comment:
It is deprecated since 1.1, which will be the most likely next release. It will be supported until 1.2 or 2.0 depending on the result of the mailing list conversation.